### PR TITLE
Fix tensorboard logger

### PR DESCRIPTION
**Pull request by TParcollet**
Our current tensor board logger is not DDP compliant and can't be resumed if the resume crash or if we restart from a different epoch. This PR fixes all this.

### CODE REVIEW

**Review by pplantinga — COMMENTED** [review]
Haven't tried this myself, but the code looks good other than my one comment.

File: recipes/DNS/enhance/spectral_map/train.py [hunk 1]
```diff
@@ -237,6 +237,7 @@ def eval_pipeline(wav, target):
         hparams["hparams"]["tensorboard_train_logger"] = TensorboardLogger(
             hparams["tensorboard_logs"]
         )
+        hparams["tensorboard_train_logger"].prepare_tensorboard_logger()
```
[the remark this answers is not in the document]
Comment (TParcollet) [reply recipes/DNS/enhance/spectral_map/train.py:240]: Yes. This is exactly why I separated this. When you initialise the SummaryWritter of tensor board, it tries to create the directory -> as it is initialised with the YAML, then DDP crashes due to multiple processes trying to create the same dir :-(

[the remark this answers is not in the document]
Comment (pplantinga) [reply recipes/DNS/enhance/spectral_map/train.py:240]: Aha, that makes sense.  Though to me it looks like all the cases listed here initialize not in YAML but rather after DDP has started. If the goal is to make this class initializable via YAML, would it make sense to move the definitions there? Even if so, perhaps that's low priority.

**Review by pplantinga — APPROVED** [review]

**Comment by mravanelli** [comment]
@TParcollet can we merge it or it is better to do another review here?

**Comment by TParcollet** [comment]
This changes quite a lot of recipe. It needs to be tested again on all of them (I did mine, but it must be reviewed).

**Comment by mravanelli** [comment]
@anautsch, could you do it when you have time? It could be great to add this fix in the new minor version.

File: recipes/DNS/enhance/spectral_map/train.py [hunk 2]
```diff
@@ -237,6 +237,7 @@ def eval_pipeline(wav, target):
         hparams["hparams"]["tensorboard_train_logger"] = TensorboardLogger(
```
Comment (anautsch) [inline recipes/DNS/enhance/spectral_map/train.py:237]: note, other scripts have here `hparams["train_logger"] = TensorboardLogger(hparams["train_log"])`

Comment (anautsch) [inline recipes/DNS/enhance/spectral_map/train.py:237]: `hparams["hparams"]` ?

**Comment by anautsch** [comment]
works with extra dependency: `protobuf==3.20.1`

protobuf is for de/serializing data structures (from Google since 2008; open-source). Going through logs which crashed, they explicitly asked for going to a 20.x protobuf version or earlier.

note: when heading for v0.6 we have a dependency & extra_dependency 'theme park' ahead of us...

**Comment by TParcollet** [comment]
This PR is not compatible with the state of the code anymore. It could be used to create a new one if someone wants to fix it.

**Pull request closed without merging**